src/tests/functional/plugin/cpu/single_layer_tests/grid_sample.cpp

```diff
@@ -0,0 +1,343 @@
+}
+
+namespace {
+
+const std::vector<ElementType> dataPrecision = {
+        ElementType::f32,
+        ElementType::bf16,
+        ElementType::i32,
+        ElementType::i8
+};
+
+const std::vector<ElementType> gridPrecision = {
+        ElementType::f32,
+        ElementType::bf16,
+        ElementType::f16
+};
+
+std::vector<std::map<std::string, std::string>> additionalConfig
+    = {{{InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::NO}},
+       {{InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::YES}}};
+
+std::vector<GridSample::InterpolationMode> interpolateMode {
+        GridSample::InterpolationMode::BILINEAR,
+        GridSample::InterpolationMode::BICUBIC,
+        GridSample::InterpolationMode::NEAREST };
+
+std::vector<GridSample::PaddingMode> paddingMode {
+        GridSample::PaddingMode::ZEROS,
+        GridSample::PaddingMode::BORDER,
+        GridSample::PaddingMode::REFLECTION };
+
+std::vector<bool> alignCorners { true, false };
+
+std::vector<CPUSpecificParams> getCPUInfo() {
+    std::vector<CPUSpecificParams> resCPUParams;
+    if (InferenceEngine::with_cpu_x86_avx512f()) {
+        resCPUParams.push_back(CPUSpecificParams{{}, {}, {"jit_avx512"}, "jit_avx512"});
+    } else if (InferenceEngine::with_cpu_x86_avx2()) {
+        resCPUParams.push_back(CPUSpecificParams{{}, {}, {"jit_avx2"}, "jit_avx2"});
+    } else if (InferenceEngine::with_cpu_x86_avx()) {
+        resCPUParams.push_back(CPUSpecificParams{{}, {}, {"jit_avx"}, "jit_avx"});
+    } else if (InferenceEngine::with_cpu_x86_sse42()) {
+        resCPUParams.push_back(CPUSpecificParams{{}, {}, {"jit_sse42"}, "jit_sse42"});
+    }
+    return resCPUParams;
+}
+
+std::vector<std::vector<InputShape>> getStaticShapes() {
+    // SSE42
+    std::vector<std::vector<InputShape>> result = {
+        { { {}, { {1, 5, 1, 1} } },   // Static shapes
+          { {}, { {1, 1, 1, 2} } }
+        },
+        { { {}, { {2, 4, 7, 1} } },   // Static shapes
+          { {}, { {2, 1, 2, 2} } }
+        },
+        { { {}, { {3, 3, 3, 3} } },   // Static shapes
+          { {}, { {3, 3, 1, 2} } }
+        },
+        { { {}, { {4, 2, 5, 4} } },   // Static shapes
+          { {}, { {4, 2, 2, 2} } }
+        },
+        { { {}, { {5, 1, 5, 5} } },   // Static shapes
+          { {}, { {5, 1, 5, 2} } }
+        },
+        { { {}, { {4, 2, 4, 6} } },   // Static shapes
+          { {}, { {4, 2, 3, 2} } }
+        },
+        { { {}, { {3, 3, 5, 7} } },   // Static shapes
+          { {}, { {3, 7, 1, 2} } }
+        },
+        { { {}, { {2, 4, 7, 7} } },   // Static shapes
+          { {}, { {2, 2, 4, 2} } }
+        },
+        { { {}, { {2, 5, 8, 8} } },   // Static shapes
+          { {}, { {2, 3, 3, 2} } }
+        },
+        { { {}, { {2, 6, 9, 8} } },   // Static shapes
+          { {}, { {2, 2, 5, 2} } }
+        }
+    }; // SSE42
+
+    if (InferenceEngine::with_cpu_x86_avx2()) {
```

Comment:
Could you tell me please what difference between shapes for ISA is? If it's last data dimension could we test Shape `[1,16,16,16]` on AVX2 as Input for example?

Reply:
There are already covered all corner cases for each ISA. [1,16,16,16] - is excessive.